templates/terraform/flatten_property_method.erb

```diff
@@ -96,9 +96,16 @@ func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d
 	if strVal, ok := v.(string); ok {
 		if intVal, err := strconv.ParseInt(strVal, 10, 64); err == nil {
 			return intVal
-		} // let terraform core handle it if we can't convert the string to an int.
+		}
 	}
-	return v
+
+	// number values are represented as float64
```

Comment:
The patchencoder for the fine-grained resource deep-compared the returned value, including the type. Terraform stores `TypeInt` as `int`, so we need to return that to compare properly. Before, we passed the value onto TF core where it did the conversion.